### DISCUSSION TITLE
Fix for an issue with Node Dock

### DISCUSSION THREAD

**Discussion by dnoftsger** [godotengine/godot-proposals#8564] [Editor]
My partner and I were taking a look at this specific issue: https://github.com/godotengine/godot/issues/84530

We believe we have an idea of how to fix this issue, and was looking for input or ideas from the developers on if we are on the right track.

We think that the issue lies in node_dock.cpp (https://github.com/godotengine/godot/blob/master/editor/node_dock.cpp), specifically with the destructor, as it currently does not undo many of the changes that occur when you create one, which would possibly be the cause for the issue linked above, in which you can still make changes to the groups and such on a node that no longer exists. We think that adding some lines to this destructor would help fix this issue, let us know what you think.

**Comment by kevmorg** [comment]
We have managed to get the windows for signals and groups to hide themselves at least on confirming the deletion of a node. We do this from the _delete_confirm function in scene_tree_dock.cpp on line 2310. Now we just need to locate where we can clear the actual signals and groups from a node to fix the issue.

**Comment by AThousandShips** [comment]
Please discuss development on the [contributor chat](https://chat.godotengine.org/), this is not for this kind of discussion but for discussing new features 🙂

**Reply by dnoftsger** [reply]
Apologies, thanks for showing us where to go

**Reply by AThousandShips** [reply]
No worries 🙂
